src/qt/askpassphrasedialog.cpp

```diff
@@ -70,6 +70,9 @@ AskPassphraseDialog::AskPassphraseDialog(Mode _mode, QWidget *parent, SecureStri
     connect(ui->passEdit3, &QLineEdit::textChanged, this, &AskPassphraseDialog::textChanged);
 
     GUIUtil::handleCloseWindowShortcut(this);
+
+    setAttribute(Qt::WA_DeleteOnClose);
```

Comment:
For the record, I've suggested to add `GUIUtil::showModalDialogAndDeleteOnClose` instead.

Reply:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/336#issuecomment-856061619).